apollo-client/src/main/java/com/ctrip/framework/apollo/internals/DefaultConfig.java

```diff
@@ -58,8 +59,9 @@ private void initialize() {
       updateConfig(m_configRepository.getConfig(), m_configRepository.getSourceType());
     } catch (Throwable ex) {
       Tracer.logError(ex);
-      logger.warn("Init Apollo Local Config failed - namespace: {}, reason: {}.",
-          m_namespace, ExceptionUtil.getDetailMessage(ex));
+      String logWarnMsg = "Init Apollo Local Config failed - namespace: {}, reason: {}.";
+      logger.warn(logWarnMsg, m_namespace, ExceptionUtil.getDetailMessage(ex));
+      DeferredLogUtil.warn(logger, logWarnMsg, m_namespace, ExceptionUtil.getDetailMessage(ex));
```

Comment:
Writing the `DeferredLogUtil.xxx` statements in many places don't seem like a very good way since it adds duplication and developer would probably forget to call this method in the future.

How about we wrap the original logger so that we don't need to duplicate the log statements? e.g.

```java
private static final Logger logger = DeferredLogger.wrap(LoggerFactory.getLogger(DefaultConfig.class));
```

or even

```java
private static final Logger logger = DeferredLoggerFactory.getLogger(DefaultConfig.class));
```

Reply:
This is a good idea

Reply:
After looking at it, it is not very easy to achieve. How about this?
```java
    public static void debug(Logger logger, String message, Object... objects) {
        logger.debug(message,objects);
        add(logger, Level.DEBUG, message, objects, null);
    }
```
Use it like this：
```java
      DeferredLogUtil.error(logger, "Initialize DefaultServerProvider failed.", ex);    
```